.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,68 @@
+name: Clang Format Check
+on:
+  pull_request:
+    paths:
+      - 'Sources/**'
+      - 'Tests/**'
+      - '.github/workflows/clang-format.yml'
+      - '.clang-format-ignore'
+
+jobs:
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install clang-format-18
+      run: |
+        wget https://apt.llvm.org/llvm.sh
+        chmod +x llvm.sh
+        sudo ./llvm.sh 18
+        sudo apt-get install -y clang-format-18
+
+    - name: Check formatting
+      run: |
+        if [ -f ".clang-format-ignore" ]; then
+          echo "Using .clang-format-ignore file to exclude files from formatting check."
+          find Sources Tests -name '*.c' -o -name '*.cpp' -o -name '*.h' -o -name '*.m' | \
+          grep -v -f .clang-format-ignore | \
```

Comment:
Why do you filter files manually here? Isn't it [a built-in function of clang-format](https://clang.llvm.org/docs/ClangFormat.html#clang-format-ignore)?